### PR TITLE
Add some functions to FEEvaluation

### PR DESCRIPTION
**Pull request by peterrum**

### CODE REVIEW

File: include/deal.II/matrix_free/fe_evaluation.h [hunk 1]
```diff
@@ -230,6 +230,18 @@ class FEEvaluationBaseData
...
+   * Return the number of the quadrature formula of the present cell.
+   */
+  unsigned int
+  get_quadrature_index() const;
+
+  /**
+   * Return index of cell or face.
+   */
+  unsigned int
+  get_index() const;
```
Comment (kronbichler) [inline include/deal.II/matrix_free/fe_evaluation.h:243]: Can we call this `get_current_cell_index()` or something similarly to indicate that it is the index we look at? I agree there is ambiguity due to the cell or face behavior defined in a derived class.

Comment (peterrum) [reply include/deal.II/matrix_free/fe_evaluation.h]: Done.

**Comment by peterrum** [comment]
@kronbichler May I ask you - once this PR is merged - to rebase #10927 so that we have a single branch for internal testing of some applications.

**Review by kronbichler — APPROVED** [review]

**Comment by kronbichler** [comment]
Thanks!

**Pull request merged**